server/aesdsocket.c

```diff
@@ -0,0 +1,363 @@
+        }
+    }
+
+
+    //listen for incoming connections. 
+    status = listen(sock_fd, BACKLOG_VAL);
+    if(status < 0)
+    {
+        printf("Error with listen(). Error code: %d\n", errno);
+        return -1;
+    }
+
+    //permissions for this file is usr=rw, g=rw, oth = r
+    filetotest_fd = open("/var/tmp/aesdsocketdata", O_RDWR | O_CREAT | O_TRUNC, S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
+
+    if(filetotest_fd < 0)
+    {
+        syslog(LOG_ERR, "Error opening the file in this location with code : %d", errno);
+    }
+
+    //Creating receive buffer to recieve and store incoming packets
+    char buffer[1024];
+
+    //Allocate storage buffer
+    packet_buffer = (char *) malloc(1024);
+
+    int packet_buffer_len = 1024;
+
+    //String to hold IP address of client from the inet_nto fn
+    char address_str[INET_ADDRSTRLEN];
+
+    //Variable to store returned string from receive
+    int bytes_received;
+
+    //Set this flag when \n is received to terminate
+    int nullterm_flag;
+
+    //Temporary variable inside the for loop for \n ind.
+    int nullchar_it = 1; 
+
+    //Variable to track the size of the packet buffer. Value of this multiplied by the standard size 1024 used when computing bytes to be written and memcpy operation on the packet buffer. 
+    int packet_buffer_size_count = 1;
+
+    //Variable to indicate number of bytes of data to be appended to the file in /var/tmp/
+    int write_len = 0;
+
+    //Variable to check the file size after writes are performed
+    int file_size = 0;
+
+    struct sockaddr_storage accept_addr;
+
+    socklen_t accept_addr_len = sizeof(accept_addr);
+    
+
+    while(1)
+    {
+        acc_fd = accept(sock_fd, (struct sockaddr *)&accept_addr, &accept_addr_len); //Accept incoming connections in a forever loop as specificed by instructions. 
+
+        if(acc_fd < 0)
+        {
+            perror("accept");
+            continue; //Continue trying to accept if failed
+        }
+
+        //Attempting to print the IP address of the accepted client connection on success of accept using the inet_ntop function
+        struct sockaddr_in *ptr = (struct sockaddr_in *)&accept_addr;
+        syslog(LOG_DEBUG, "Accepted connection from IP : %s", inet_ntop(AF_INET, &ptr->sin_addr, address_str, sizeof(address_str)));
+
+        //Receive all bytes from client
+        while((bytes_received = recv(acc_fd, buffer, sizeof(buffer), 0)) > 0)
+        {
+            printf("Verifying the number of bytes received: %d\n", bytes_received);
+
+            //Check for \n for the end of a packet received. 
+            for(int i = 0; i < bytes_received; i++)
+            {
+                if(buffer[i] == '\n')
+                {
+                    nullterm_flag = 1;
+                    nullchar_it = i+1;
+                    break;
+                }
+            }
+
+            memcpy(packet_buffer + (packet_buffer_size_count -1)*1024, buffer, bytes_received);
```

Comment:
@suja8742 I don't think this logic will work when a buffer is < 1024 bytes but doesn't contain a `\n`.  I think you'll have a bunch of uninitialized data between two `recv` buffers.

Reply:
Thanks for pointing it out, handling this.